In [1]:
import pandas as pd 
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from scipy import stats

## Q 1) 1.	The time required for servicing transmissions is normally distributed with  = 45 minutes
## and = 8 minutes. The service manager plans to have work begin on the transmission of a customer’s car 
## 10 minutes after the car is dropped off and the customer is told that the car will be ready within 1 hour from drop-off.
## What is the probability that the service manager cannot meet his commitment? 

### A.	0.3875   
### B.	0.2676   
### C.	0.5   
### D.	0.6987 


## Ans:
### The work begin at 10 min. after the car droped off.The maneger committed to customers that
### work will be completed within 1 hr.
### So the time left to do the work is 50 min.
### The probability that maneger can not meet his commitement = P(X>50)

In [2]:
## P value
## X=50,mu=45,sigma=8
stats.norm.cdf(50,45,8)

0.7340144709512995

### P(X<50)=0.7340
### P(X>50)=1-P(X<50)
###        = 1-0.7340
###        = 0.2676
## So the answer is B

## Q2) 2.	The current age (in years) of 400 clerical employees at an insurance claims processing center is 
## normally distributed with mean  = 38 and Standard deviation  =6. For each statement below, 
## please specify True/False. If false, briefly explain why.
### A.	More employees at the processing center are older than 44 than between 38 and 44.
### B.	A training program for employees under the age of 30 at the center would be expected to attract about 36 employees.



## Ans:
### i) Probability of employee older than 44 P(X>44)
### &  Probability of employee betweeen 38 &  44  is P(38<X<44)


In [3]:
## P(X<44)
p_value=stats.norm.cdf(44,38,6)

In [4]:
p_value

0.8413447460685429

In [5]:
## P(38<X<44)
stats.norm.cdf(44,38,6)-stats.norm.cdf(38,38,6)

0.3413447460685429

#### P(X<44) >  P(38<X<44)
#### Hence the statment is true.

In [6]:
### ii) P(X<30)
stats.norm.cdf(30,38,6)

0.09121121972586788

#### Number of employee with probability 0.091211 = 400*0.091211 = 36.484
#### Hence The Statment is True.

## Q.3) 3.	If X1 ~ N(μ, σ2) and X2 ~ N(μ, σ2) are iid normal random variables, 
## then what is the difference between 2 X1 and X1 + X2? Discuss both their distributions and parameters.       



## Ans:
#### We know that X ~ N(μ, σ2) and Y~ N(μ, σ2) are iid normal random variables,
#### then X+Y ~ N(μ1+μ2, (σ1)2+ (σ2)2) 
#### Also Z=aX+bY , X & Y are defined as above then Z~N(aμ1+bμ2, a2(σ1)2+b2(σ2)2)


#### Using above, we can find that 2X1~N(2μ,2σ2) & X1+X2~N(μ+μ,σ2+σ2)
####                                               X1+X2~N(2μ,2σ2)
#### 2X1-(X1+X2)~N(4μ,4σ2)

## Q.4) Let X ~ N(100, 202). Find two values, a and b, symmetric about the mean, 
## such that the probability of the random variable taking a value between them is 0.99. 

### A.	90.5, 105.9 
### B.	80.2, 119.8 
### C.	22, 78 
### D.	48.5, 151.5 
### E.	90.1, 109.9



In [9]:
## Ans:
stats.norm.interval(0.99,100,20)

(48.48341392902199, 151.516586070978)

### Answer is D

## Q.5) 5.	Consider a company that has two different divisions.
## The annual profits from the two divisions are independent and have distributions 
## Profit1 ~ N(5, 32) 
## and Profit2 ~ N(7, 42) respectively. Both the profits are in  Million.
## Answer the following questions about the total profit of the company in Rupees. Assume that 1 = Rs. 45
### A.	Specify a Rupee range (centered on the mean) such that it contains 95% probability for the annual profit 
### of the company.
### B.	Specify the 5th percentile of profit (in Rupees) for the company
### C.	Which of the two divisions has a larger probability of making a loss in a given year?


## Ans:
### i)
#### mean=5+7=12
#### mean profit in rupees = 12*45 = 540 million
#### S.D = sqrt(9+16)=sqrt(25)=5
#### In RS, S.D = 5*45= 225 million

In [11]:
stats.norm.interval(0.95,540,225)

(99.00810347848784, 980.9918965215122)

#### Rupee range  for the annual profit is (99.00810347848784, 980.9918965215122)

### ii)

In [13]:
#### X=(μ+zσ) 
540+(-1.645)*(225)

169.875

In [14]:
## iii) Probability of division making a loss is P(x<0):

In [15]:
stats.norm.cdf(0,5,3)

0.0477903522728147

In [16]:
stats.norm.cdf(0,7,4)

0.040059156863817086